slack/handler.go

```diff
@@ -83,25 +79,83 @@ func (r *rtmBot) handleEvent(msg slack.RTMEvent) error {
 		return errors.New("Invalid auth received")
 	case *slack.HelloEvent:
 		r.logger.Println("Received hello, sending greetings !")
-		r.rtm.SendMessage(r.rtm.NewOutgoingMessage("Hello, I'm musicof, let's play !", r.channel.ID))
+		r.rtm.PostMessage(r.channel.ID, "Hello, I'm musicof, let's play !", slack.PostMessageParameters{})
 	case *slack.ConnectedEvent:
 		r.logger.Println("Connected !")
 	case *slack.MessageEvent:
 		if err := r.handleMessage(ev); err != nil {
+			r.logger.Println("Failed to handle message, reason :", err)
 			return err
 		}
 	}
 
 	return nil
-
 }
 
-func (r *rtmBot) handleHalt() {
-	r.rtm.SendMessage(r.rtm.NewOutgoingMessage("See you later !", r.channel.ID))
+func (r *rtmBot) handleHalt() error {
+	r.rtm.PostMessage(r.channel.ID, "See you later !", slack.PostMessageParameters{})
+	return r.rtm.Disconnect()
 }
 
 func (r *rtmBot) handleMessage(ev *slack.MessageEvent) error {
-	// TODO write something interesting
+	if ev.Channel != r.channel.ID {
+		return nil
+	}
 
-	return nil
+	if !strings.Contains(ev.Text, r.rtm.GetInfo().User.ID) {
+		return nil
+	}
+
+	if !strings.Contains(ev.Text, "nominate") {
+		return nil
+	}
+
+	return r.handleNominate(ev.User)
+}
+
+func (r *rtmBot) handleNominate(callerID string) error {
+	botID := r.rtm.GetInfo().User.ID
+	unwantedIDs := []string{botID, callerID}
+
+	usersInConversationParameters := &slack.GetUsersInConversationParameters{ChannelID: r.channel.ID}
+	userIDs, _, err := r.rtm.GetUsersInConversation(usersInConversationParameters)
+
+	if err != nil {
+		return err
+	}
+
+	userIDs = remove(userIDs, unwantedIDs)
+
+	userID := userIDs[rand.Intn(len(userIDs))]
+
+	user, err := r.rtm.GetUserInfo(userID)
+
+	if err != nil {
+		return err
+	}
+
+	_, _, err = r.rtm.PostMessage(r.channel.ID, "@"+user.Name, slack.PostMessageParameters{LinkNames: 1})
+
+	return err
+}
+
+func remove(s []string, r []string) []string {
```

Comment:
You can use variadic parameters in order to simplify this function usage.

For instance : 
`func remove(s []string, r ...string) []string` 
Then you can call your function using 
`remove(slice, value1, value2)` instead of `remove(slice, []string{value1,value2})`

Reply:
Also you can rename `remove` to `filter` because this is a filtering operation ;)